File: aspnet-core/src/SmartPos.Core/Staff/StaffPayroll.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Abp.Domain.Entities;
using Abp.Domain.Entities.Auditing;
using SmartPos.Branches;

namespace SmartPos.Staffs
{
    [Table("AppStaffPayrolls")]
    public class StaffPayroll : FullAuditedEntity, IMayHaveTenant
    {
        public const int MaxRemarksLength = 512;

        public virtual int? TenantId { get; set; }

        public virtual int BranchId { get; set; }

        [ForeignKey(nameof(BranchId))]
        public virtual Branch Branch { get; set; }

        public virtual int StaffId { get; set; }

        [ForeignKey(nameof(StaffId))]
        public virtual Staff Staff { get; set; }

        public virtual int Month { get; set; }

        public virtual int Year { get; set; }

        public virtual decimal BasicSalary { get; set; }

        public virtual decimal Allowance { get; set; }

        public virtual decimal Bonus { get; set; }

        public virtual decimal Deduction { get; set; }

        public virtual decimal OvertimeAmount { get; set; }

        public virtual decimal NetSalary { get; set; }

        public virtual PayrollPaymentStatus PaymentStatus { get; set; }

        public virtual DateTime? PaymentDate { get; set; }

        [StringLength(MaxRemarksLength)]
        public virtual string Remarks { get; set; }
    }
}

File: angular/src/app/demo/api/staff-payroll.ts
export enum PayrollPaymentStatus {
    Pending = 0,
    Paid = 1,
    Cancelled = 2,
}

export interface StaffPayrollDto {
    id: number;
    branchId: number;
    branchName?: string;
    staffId: number;
    staffName?: string;
    month: number;
    year: number;
    basicSalary: number;
    allowance: number;
    bonus: number;
    deduction: number;
    overtimeAmount: number;
    netSalary: number;
    paymentStatus: PayrollPaymentStatus;
    paymentDate?: string | Date | null;
    remarks?: string;
}

export interface CreateStaffPayrollDto {
    staffId: number;
    month: number;
    year: number;
    basicSalary?: number | null;
    allowance: number;
    bonus: number;
    deduction: number;
    overtimeAmount: number;
    paymentStatus: PayrollPaymentStatus;
    paymentDate?: string | Date | null;
    remarks?: string;
}

export interface PagedStaffPayrollResultRequestDto {
    keyword?: string;
    staffId?: number | null;
    month?: number | null;
    year?: number | null;
    paymentStatus?: PayrollPaymentStatus | null;
    skipCount?: number;
    maxResultCount?: number;
}
